<a href="https://colab.research.google.com/github/ana-fav/TheLook_SalesReturn_Prediction/blob/main/TheLook_ABT_Safras.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Importanto bibliotecas
import os
import numpy as np
import pandas as pd
import dateutil

In [5]:
# Carregando os dados

# Diretórios para o carregamento do Dataset
pasta_raiz = '/content/drive/MyDrive/Portifólio Data Science/datasets'
DATA_DIR = os.path.join(pasta_raiz, 'TheLook_ecommerce')

## 1.1. Tabela Pedidos

In [7]:
# Tabela de pedidos

df_orders = pd.read_csv(os.path.join(DATA_DIR, 'orders.csv'), parse_dates=['created_at', 'shipped_at', 'delivered_at'])
df_orders.head()

order_id  user_id   status gender                created_at returned_at  \
0         1        1  Shipped      F 2022-07-07 16:16:00+00:00         NaN   
1         2        1  Shipped      F 2022-12-21 16:16:00+00:00         NaN   
2         8        6  Shipped      F 2023-03-09 11:46:00+00:00         NaN   
3         9        6  Shipped      F 2022-11-20 11:46:00+00:00         NaN   
4        13        9  Shipped      F 2022-12-01 01:42:00+00:00         NaN   

                 shipped_at delivered_at  num_of_item  
0 2022-07-07 17:02:00+00:00          NaT            2  
1 2022-12-23 00:01:00+00:00          NaT            1  
2 2023-03-11 03:05:00+00:00          NaT            1  
3 2022-11-23 09:48:00+00:00          NaT            4  
4 2022-12-02 03:05:00+00:00          NaT            1

In [9]:
# Verificando a quantidade total de pedidos e de clientes
print(f'Qtde. Pedidos: {df_orders["order_id"].nunique()}')
print(f'Qtde. Clientes: {df_orders["user_id"].nunique()}')

Qtde. Pedidos: 124084
Qtde. Clientes: 79743


In [10]:
# Verificando o intervalo de datas na base
df_orders.agg({'created_at': ['min', 'max']})

created_at
min        2019-01-06 11:16:00+00:00
max 2023-04-10 19:40:25.434338+00:00

In [11]:
# Explorando os status das vendas
df_orders['status'].value_counts()

Shipped       37330
Complete      30924
Processing    24627
Cancelled     18689
Returned      12514
Name: status, dtype: int64

In [12]:
# Explorando os status das vendas de forma normalizada
df_orders['status'].value_counts(normalize=True)

Shipped       0.300845
Complete      0.249218
Processing    0.198470
Cancelled     0.150616
Returned      0.100851
Name: status, dtype: float64

Temos uma base com dados de aproximadamente 4 anos, com um total de 124.084 vendas realizadas, para um total de 79743 clientes

A base conta com 5 diferentes status de pedidos, sendo que 24,9% dos pedidos possuem o status "Complete" e 10,1% o status 'Returned'


## 1.2. Itens

In [14]:
df_order_items = pd.read_csv(os.path.join(DATA_DIR, 'order_items.csv'), parse_dates=['created_at', 'shipped_at', 'delivered_at', 'returned_at'])
df_order_items.head()

id  order_id  user_id  product_id  inventory_item_id      status  \
0   67672     46807    37876       13606             182487     Shipped   
1   64687     44719    36189       13606             174393   Cancelled   
2  102665     70878    57088       13606             276935  Processing   
3   86714     59861    48162       28951             233783     Shipped   
4  156776    108213    87198       28951             423048     Shipped   

                 created_at                shipped_at delivered_at  \
0 2022-09-24 03:30:22+00:00 2022-09-21 07:13:00+00:00          NaT   
1 2022-03-30 04:22:22+00:00                       NaT          NaT   
2 2022-09-29 07:40:37+00:00                       NaT          NaT   
3 2021-01-13 14:25:39+00:00 2021-01-14 15:38:00+00:00          NaT   
4 2021-08-03 22:27:03+00:00 2021-08-04 22:51:00+00:00          NaT   

  returned_at  sale_price  
0         NaT         2.5  
1         NaT         2.5  
2         NaT         2.5  
3         NaT         3.0  
4         NaT         3.0

In [16]:
# Verificando a quantidade total de produtos
print(f'Qtde. Produtos: {df_order_items["product_id"].nunique()}')

Qtde. Produtos: 29039


In [17]:
# Verificando o intervalo de na preços dos protudos e dos fretes
df_order_items['sale_price'].agg(['min', 'max'])

min      0.02
max    999.00
Name: sale_price, dtype: float64